Dockerfile

```diff
@@ -107,7 +107,7 @@ RUN patch -d / -p0 < /tmp/skip_key_copy.patch
 # Fix Ubuntu20 OS detection (SOFTWARE-4463)
 # Can be dropped when HTCONDOR-242 is involved
 COPY hosted-ce/overrides/HTCONDOR-242.remote-os-detection.patch /tmp
-RUN [[ $BASE_YUM_REPO == 'development' ]] || patch -d / -p0 < /tmp/HTCONDOR-242.remote-os-detection.patch
+RUN [[ $BASE_YUM_REPO != 'release' ]] || patch -d / -p0 < /tmp/HTCONDOR-242.remote-os-detection.patch
```

Comment:
I am a bit confused what's going on here.

You want to apply the patch iff installing from the main osg release repo, because the version of condor in osg-testing and osg-development is 9.0.0, against which this patch does not apply?

Reply:
Correct! I guess I could switch out `||` for `&&` to make this clearer if you'd prefer

Reply:
Just wanted to check.  I don't have a strong preference between the `&&` and `||`.

In this case, 

```
[[ $BASE_YUM_REPO = 'release' ]] && patch ...
```

might be slightly more readable (without squinting), but i am always a bit more wary of using `&&` because the left hand side being false (as would be the case for `development` or `testing`) results in the whole line returning a failure status.  And in that case i think the `RUN` command would interpret the command as a failure.

`||` has the advantage that the line only returns failure if the command on the right hand side fails (eg, `patch` fails to apply).

...

Apart from which operator you use though, it just wasn't obvious to me looking at the PR what the intention was and which osg repos were supposed to have the (apparently unpatchable) condor 9.0.0.

Did the patch/fix make it into upstream 9.0.0 then?  Or do we want to make a different patch for 9.0.0?

Reply:
Yup, the patch made it into upstream 8.9.12 (which we didn't ship in the OSG): https://opensciencegrid.atlassian.net/browse/HTCONDOR-242.

I thought about always applying the patch `... ||:` but I do kinda like keeping it around so we remember to get rid of it when we finally push 9.0.0 through release